src/server/project.ts

```diff
@@ -1343,7 +1354,15 @@ namespace ts.server {
                 });
                 this.lastReportedFileNames = currentFiles;
                 this.lastReportedVersion = this.projectProgramVersion;
-                return { info, changes: { added, removed, updated }, projectErrors: this.getGlobalProjectErrors() };
+                return {

```

Comment:
Shouldn't you include file in updated if it was already reported but whether it is source of project reference redirect or not has changed?

Reply:
That is an excellent point that I would never have thought of.

Reply:
Now tracking in additional response property `updatedRedirects`.
It didn't make sense to combine into `updated` because the only way to determine from the editor which members of `updated` were present due to a redirect info change and which were present due to the existing meaning was to rely on the implicit invariant that the latter category had non-traditional file extensions. In short... it was clearer this way. :)